test/unit/extension/lib/catalog/product/service/ShopgateImageBuilderTest.js

```diff
@@ -0,0 +1,67 @@
+const ShopgateImageBuilder = require('../../../../../../../extension/lib/catalog/product/service/ShopgateImageBuilder')
+const assert = require('assert')
+
+describe('Shopgate Image Builder > ', () => {
+  let tempProduct = (tempVariant = {})
+  beforeEach(() => {
+    tempProduct = { id: 1, images: [], variants: [] }
+    tempVariant = { id: 2, image_url: '' }
+  })
+  describe('Parent > ', () => {
+    it('Returns parent image array if entries are not empty', function() {
+      const expectedImages = ['1.jpg', '2.jpg']
+      tempProduct.images = [{ url_standard: '1.jpg' }, { url_standard: '' }, { url_standard: '2.jpg' }]
+
+      const returns = new ShopgateImageBuilder(tempProduct, 0).build()
+
+      assert.deepEqual(returns.images, expectedImages)
+    })
+
+    it('Returns and empty array if there are no parent images', function() {
+      const returns = new ShopgateImageBuilder(tempProduct, 0).build()
+
+      assert.deepEqual(returns.images, [])
+    })
+  })
+
+  describe('Variants > ', () => {
+    it("Returns parent's images if no variants exist", function() {
+      const expectedImages = ['1.jpg', '2.jpg']
+      tempProduct.images = [{ url_standard: '1.jpg' }, { url_standard: '2.jpg' }]
+
+      const returns = new ShopgateImageBuilder(tempProduct, 2).build()
+
+      assert.deepEqual(returns.images, expectedImages)
+    })
+
+    it("Returns parent's images if no variant images are set", function() {
```

Comment:
use single quotes here. same for the following test descriptions

Reply:
it's just the auto formatter, it changes to double quotes because there is a single quote inside it to avoid needing to add ugly escapes to those. But I will rephrase to not have single quotes inside.